project1/application.py

```diff
@@ -29,15 +40,33 @@ def index():
 @app.route("/register",methods=["POST","GET"])
 def register():
     # session.clear()
+    # id11 = 100
     if request.method == "GET":    
         return render_template("register.html")
-        # username = request.form.get("username")
-        # pass1 = request.form.get("pass")
-        # print(username+"    "+pass1)
+
     if request.method == "POST":
-        # print("here")
+        # id = id11
         username = request.form.get("username")
+        mailid = request.form.get("mail_id")
         pass1 = request.form.get("password")
-        print (username+"    "+pass1)
-        names[username] = pass1
-        return render_template("userlist.html",name = names.keys())
+        date_time = datetime.datetime.now()
+        emails = user.query.filter_by(mail_id = mailid).first()
+        if emails is None:
+            new_user = user(user_name = username, mail_id = mailid, password = pass1,date_time=date_time)
+            db.session.add(new_user)
+            db.session.commit()
+            # id11 += 1/
+            return render_template("sucess.html")
+        else:
+            return render_template("error.html", msg= "Email is already registered")
+
+@app.route("/admin")
+def admin():
+    data = user.query.order_by(user.user_name).all()
+    # print(data)
+    data1=[]
+    for i in data:
+        print(i.user_id,i.user_name,i.date_time)
+        data1.append([i.user_name,i.date_time])
+    return render_template("admin.html",datas=data1)
+
```

Comment:
Instead of array, send the data to the html page & loop there. You can use i.user_id like you did here in the html page too.

Reply:
done

Reply:
left a comment click on view changes to refer to the  modifications